src/org/loklak/api/search/QuoraProfileScraper.java

```diff
@@ -76,10 +79,176 @@ public JSONObject getDefaultPermissions(BaseUserRole baseUserRole) {
         return new HashMap<String, String>();
     }
 
-    private Post scrapeProfile() {
+    @Override
+    public Timeline2 getData() {
+        //TODO: 1) to get this 2) convert to array
+        //String type = this.extra.get("type");
+        String type = "all";
+        String midUrl;
+        String url;
+        Thread[] dataThreads = new Thread[3];
+        this.postList = new Timeline2(this.order);
+
+        switch(type) {
+            case "all":
+            case "profile":
+                midUrl = "profile/";
+                url = this.baseUrl + midUrl + this.query;
+                dataThreads[0] = new ConcurrentScrape(url, "profile");
+                dataThreads[0].start();
+                if("profile".equals(type)) break;
+            case "question":
+                midUrl = "search/?q=";
+                //TODO: use request body instead
+                url = this.baseUrl + midUrl + this.query + "&type=question";
+                dataThreads[1] = new ConcurrentScrape(url, "question");
+                dataThreads[1].start();
+                if("question".equals(type)) break;
+            case "answer":
+                midUrl = "search/?q=";
+                //TODO: use request body instead
+                url = this.baseUrl + midUrl + this.query + "&type=answer";
+                dataThreads[2] = new ConcurrentScrape(url, "answer");
+                dataThreads[2].start();
+                if("answer".equals(type)) break;
+            default:
+                break;
+        }
+
+        try {
+            for (int i=0; i<3; i++) {
+                dataThreads[i].join();
+            }
+        } catch(InterruptedException e) {
+            DAO.severe("Couldn't complete all threads");
```

Comment:
Maybe we need to mention the scraper name and `type` variable here so that it is easy to debug.